backend/danswer/connectors/sharepoint/connector.py

```diff
@@ -75,44 +78,25 @@ def __init__(
     ) -> None:
         self.batch_size = batch_size
         self.graph_client: GraphClient | None = None
-        self.site_data = self.extract_site_and_folder(sites)
+        self.site_data = self._extract_site_and_folder(sites)
 
     @staticmethod
-    def extract_site_and_folder(site_urls: list[str]) -> list[SiteData]:
+    def _extract_site_and_folder(site_urls: list[str]) -> list[SiteData]:
         site_data_list = []
         for url in site_urls:
             parts = url.strip().split("/")
             if "sites" in parts:
                 sites_index = parts.index("sites")
-                site_url = "/".join(parts[:sites_index + 2])
-                folder = parts[sites_index + 2] if len(parts) > sites_index + 2 else None
-                site_data_list.append(SiteData(url=site_url, folder=folder, siteobjects=[], driveitems=[]))
+                site_url = "/".join(parts[: sites_index + 2])
+                folder = (
+                    parts[sites_index + 2] if len(parts) > sites_index + 2 else None
+                )
+                site_data_list.append(
+                    SiteData(url=site_url, folder=folder, siteobjects=[], driveitems=[])
+                )
         return site_data_list
 
-    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
```

Comment:
In terms of how to layout functions: probably best practice to have externally used functions at the top for ease of finding but in this repo we've generally put helper functions above the functions that use them and external functions at the bottom. Consistency is more important.

In terms of function arguments, the arguments that are most varied and critical go first, arguments that are generally the same every time (like passing in a db_session object) go at the end.